driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -151,10 +151,46 @@ public void onSuccess(List<Void> l) {
                     initFuture.set(null);
                 }
             }
+
+            @Override
+            public void onFailure(Throwable t) {
```

Comment:
Good catch: it was because the fallback in `setKeyspaceAsync` was not applied when the keyspace is null. I'm splitting that into two separate methods.